ocaml/xapi/xapi_cluster.ml

```diff
@@ -47,6 +47,29 @@ let create ~__context ~pIF ~cluster_stack ~pool_auto_join ~token_timeout ~token_
       assert_pif_prerequisites (pIF,pifrec);
       let ip = ip_of_pif (pIF,pifrec) in
 
+      let assert_pool_on_same_subnet () =
+        let are_on_same_subnet =
+          try
+            let parse_ip ip_str = Scanf.sscanf ip_str "%i.%i.%i.%i" (fun a b c d -> (a, b, c, d)) in
```

Comment:
This is clear and simple. But we also have the `Ipaddr` which provides functions for this. I'm not sure what to recommend.

Reply:
i will use that. i assume I will also be able to find a `to_network_address` function